### DISCUSSION TITLE
How to for_each with groupby?

### DISCUSSION THREAD

**Discussion by vkensou** [skypjack/entt#1189] [Q&A] [answered]
I want to batch process objects with the same material and mesh. How should I do it?
sort group and then record batch's begin/end in an extra vector? Does entt support this immediately?

**Comment by skypjack** [comment]
I guess you want to pass around all components of a given type as a packed array? Remember that EnTT storage is paged by default, you still have to provide a page at a time in case. Do you know it?

**Reply by vkensou** [reply]
> I guess you want to pass around all components of a given type as a packed array? Remember that EnTT storage is paged by default, you still have to provide a page at a time in case. Do you know it?

I want group entities by a value/a lambda, like opaque objects is group1 and transparent objects is group2, etc. Not group by component, but group by value.

**Reply by skypjack** [reply, marked as answer]
As in:
```cpp
struct clazz {
    int foo{3};
    // ...
};
```
Gimme all the entities for which `foo`is equal to 3?
No, EnTT doesn't support it out of the box unless you use something like the reactive storage and keep track of the entities of interest by checking a runtime value like this.
However, it smells of an XY-problem to me.

**Reply by vkensou** [reply]
Thank you for your reply. It seems that I need to think deeply.
